Application/src/model/Parser.java

```diff
@@ -73,43 +75,48 @@ public void parseSequenceDiagram(String inputJSON) {
+            // Assigns a String to the diagram's format
+            String format = meta.getFormat();
+            // Assigns a String to the diagram's extensions
+            List<Object> extensions = meta.getExtensions();
+            // Assigns a String to the diagram's version
+            String version = meta.getVersion();
+            // Assigns a String to the type of the diagram
+            String type = parsedSequenceDiagram.getType();
 
             // get the tempProcesses which contain the class names
-            for (Processes processesElement : parsedDiagram.getProcesses()) {
+            for (Processes processesElement : parsedSequenceDiagram.getProcesses()) {
                 tempProcesses += "\"" + processesElement.getSequenceDiagramClass() + ":" + processesElement.getName() + "\",";
                 processesString = (tempProcesses.substring(0, tempProcesses.length() - 1));
-                // yield only the names of the classes
+                // yield only the names of the Classes
                 tempClassNames += processesElement.getName() + ",";
                 //removing extra comma
                 classNamesString = (tempClassNames.substring(0, tempClassNames.length() - 1));
             }
 
             // get the elements of the First Diagram
-            for (ContentArray diagramElement : parsedDiagram.getDiagram().getContent().get(0).getContent()) {
-                tempFirstDiagram += "{" + diagramElement.getFrom() + "," + diagramElement.getTo() + ",[" + diagramElement.getMessage().get(0) + ", " + diagramElement.getMessage().get(1) + ", " + diagramElement.getMessage().get(2) + "]" + "}" + ",";
+            for (Messages diagramElement : parsedSequenceDiagram.getDiagram().getContent().get(0).getMessages()) {
+                tempFirstDiagram += "{" + diagramElement.getFrom() + "," + diagramElement.getTo() + ",[" + diagramElement.getMessage().get(0) + ", " +
+                        diagramElement.getMessage().get(1) + ", " + diagramElement.getMessage().get(2) + "]" + "}" + ",";
                 //removing extra comma
                 firstDiagramString = (tempFirstDiagram.substring(0, tempFirstDiagram.length() - 1));
             }
 
 
             // get the elements of the Parallel Diagram
-            for (ContentArray parallelDiagramElement : parsedDiagram.getDiagram().getContent().get(1).getContent()) {
+            for (Messages parallelDiagramElement : parsedSequenceDiagram.getDiagram().getContent().get(1).getMessages()) {
                 tempParallelDiagram += "{" + parallelDiagramElement.getFrom() + "," + parallelDiagramElement.getTo() + ",[" + parallelDiagramElement.getMessage().get(0) + "]" + "}" + ",";
                 //removing extra comma
                 parallelDiagramString = (tempParallelDiagram.substring(0, tempParallelDiagram.length() - 1));
             }
 
-        } catch (NullPointerException e) {
+        } catch (Exception e) {
```

Comment:
All errors are syntax error related?
Also the syntax error says 'import failed' not a 'parsing failed'?

Reply:
Not all users know what parsing is, so let's keep it at "import failed"
are there any other parsing error not related to a bad syntax/ unrecognizable syntax?